pkg/addon-operator/operator.go

```diff
@@ -2058,14 +2058,26 @@ func InitAndStart(operator *AddonOperator) error {
 
 func QueueHasModuleRunTask(q *queue.TaskQueue, moduleName string) bool {
 	hasTask := false
-	q.Filter(func(t sh_task.Task) bool {
-		if t.GetType() == task.ModuleRun {
-			hm := task.HookMetadataAccessor(t)
-			if hm.ModuleName == moduleName {
-				hasTask = true
-			}
+	firstTask := true
+
+	q.Iterate(func(t sh_task.Task) {
+		// Skip the first task in the queue.
+		// The first task is actually not in the queue anymore, because it has been already executed.
+		if firstTask {
+			firstTask = false
+			return
+		}
+
+		if t.GetType() != task.ModuleRun {
+			return
 		}
-		return true
+
+		if task.HookMetadataAccessor(t).ModuleName != moduleName {
+			return
+		}
+
+		hasTask = true
```

Comment:
Uh-oh, I don't like this "inverted logic" style ;) Is it possible to revert to a previous condition `if type == ModuleRun && name == moduleName { hasTask = true }`?